### PR TITLE
feat: implemented Book entity, BookRepository and BookService with sa…

### PR DESCRIPTION
**Pull request by PavloMelnyk10**
…ve() and findAll() methods

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/SpringBookstoreApplication.java [hunk 1]
```diff
@@ -1,12 +1,39 @@
...
+    private BookService bookService;
+
     public static void main(String[] args) {
         SpringApplication.run(SpringBookstoreApplication.class, args);
     }
 
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            Book book1 = new Book(null, "Spring Boot Essentials", "John Smith",
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/SpringBookstoreApplication.java]: don't use numbers for naming

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]: Don't use `@Data` for entities
```suggestion
```

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]: Redundant annotations 
```suggestion
```

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+public class Book {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java:17]: Add `@Table`

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java:13]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public Book save(Book book) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java]: don't use try with resources and transaction, session will be closed before catch block and rollback won't happened if exception will thrown

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Failed to add book: " + book, e);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Failed to add book: " + book, e);
```

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaQuery<Book> criteriaQuery = session.getCriteriaBuilder()
+                    .createQuery(Book.class);
+            criteriaQuery.from(Book.class);
+            return session.createQuery(criteriaQuery).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all books", e);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get all books", e);
```

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+                transaction.rollback();
+            }
+            throw new RuntimeException("Failed to add book: " + book, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaQuery<Book> criteriaQuery = session.getCriteriaBuilder()
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java]: There is no sense to use CriteriaBuilder

File: src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package mate.academy.bookstore.service.impl;
+
+import java.util.List;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.BookRepository;
+import mate.academy.bookstore.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java:11]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+public class BookServiceImpl implements BookService {
+    @Autowired
+    private BookRepository bookRepository;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
    private final BookRepository bookRepository;
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor
+public class BookRepositoryImpl implements BookRepository {
+    @Autowired
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java (hunk 11 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java:11]: add `@RequiredArgsConstructor`

File: src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java (hunk 12 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java]: fix comment

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get all books", e);
+        }
+    }
+
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java:48]:
```suggestion
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**